gtsfm/loader/mobilebrick_loader.py

```diff
@@ -0,0 +1,135 @@
+"""Simple loader class that reads from the MobileBrick dataset.
+
+Reference to MobileBrick: https://code.active.vision/MobileBrick/, Kejie Li et al.
+
+Authors: Akshay Krishnan
+"""
+
+import os
+from pathlib import Path
+from typing import List, Optional
+
+import numpy as np
+
+from gtsam import Cal3Bundler, Pose3, Rot3
+
+import gtsfm.utils.io as io_utils
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.image import Image
+from gtsfm.loader.loader_base import LoaderBase
+
+logger = logger_utils.get_logger()
+
+
+class MobilebrickLoader(LoaderBase):
+    """Loader class that reads from the MobileBrick dataset."""
+
+    def __init__(
+        self,
+        data_dir: str,
+        max_frame_lookahead: int = 5,
+        max_resolution: int = 1024,
+        input_worker: Optional[str] = None,
+    ) -> None:
+        """ """
+        super().__init__(max_resolution=max_resolution, input_worker=input_worker)
+
+        self._max_frame_lookahead = max_frame_lookahead
+        self._image_dir = os.path.join(data_dir, "image")
+        self._num_images = len(os.listdir(self._image_dir))
+
+        # Cache image paths
+        self._image_paths = []
+        for i in range(self._num_images):
+            image_path = os.path.join(self._image_dir, f"{i:06d}.jpg")
+            self._image_paths.append(image_path)
+
+        # Load GT intrinsics
+        self._intrinsics_dir = os.path.join(data_dir, "intrinsic")
+        self._intrinsics = []
+        for i in range(self._num_images):
+            intrinsics_file = os.path.join(self._intrinsics_dir, f"{i:06d}.txt")
+            K = np.loadtxt(intrinsics_file)
+            self._intrinsics.append(Cal3Bundler((K[0, 0] + K[1, 1]) / 2, 0, 0, K[0, 2], K[1, 2]))
+
+        # Load GT poses
+        self._poses_dir = os.path.join(data_dir, "pose")
+        self._wTi = []
+        for i in range(self._num_images):
+            pose_file = os.path.join(self._poses_dir, f"{i:06d}.txt")
+            wTi_mat = np.loadtxt(pose_file)
+            wTi = Pose3(Rot3(wTi_mat[:3, :3]), wTi_mat[:3, 3])
+            self._wTi.append(wTi)
+
+    def image_filenames(self) -> List[Path]:
+        """Return the file names corresponding to each image index."""
+        return [Path(fpath) for fpath in sorted(os.listdir(self._image_dir))]
+
+    def __len__(self) -> int:
+        """The number of images in the dataset.
+
+        Returns:
+            The number of images.
+        """
+        return self._num_images
+
+    def get_image_full_res(self, index: int) -> Image:
+        """Get the image at the given index, at full resolution.
+
+        Args:
+            index: The index to fetch.
+
+        Raises:
```

Comment:
nit: usually `Raises` would come after the `Returns`: https://google.github.io/styleguide/pyguide.html#244-decision